src/FakeItEasy/Core/FakeManager.ObjectMemberRule.cs

```diff
@@ -54,14 +58,29 @@ public void Apply(IInterceptedFakeObjectCall fakeObjectCall)
                 }
             }
 
+            private static bool IsSameMethod(MethodInfo first, MethodInfo second)
+            {
+                return first.DeclaringType == second.DeclaringType
+#if FEATURE_REFLECTION_METADATATOKEN
+                   && first.MetadataToken == second.MetadataToken
+#else
+                   && first.Name == second.Name
+                   && first.GetParameters().Select(p => p.ParameterType).SequenceEqual(second.GetParameters().Select(p => p.ParameterType))
+#endif
+                   && first.Module == second.Module
+                   && first.GetGenericArguments().SequenceEqual(second.GetGenericArguments());
+            }
+
             private static bool IsObjetMethod(IFakeObjectCall fakeObjectCall)
             {
-                return ObjectMethodsMethodHandles.Contains(fakeObjectCall.Method.MethodHandle);
+                return IsSameMethod(ObjectMethods[0], fakeObjectCall.Method)
+                    || IsSameMethod(ObjectMethods[1], fakeObjectCall.Method)
+                    || IsSameMethod(ObjectMethods[2], fakeObjectCall.Method);

```

Comment:
But `MetadataToken` isn't enough to uniquely identify a member, IIRC... it has to be used in combination with the `Module`.